dex/src/state.rs

```diff
@@ -2686,8 +2691,201 @@ impl State {
     }
 
     #[cfg(feature = "program")]
-    fn process_send_take(_args: account_parser::SendTakeArgs) -> DexResult {
-        unimplemented!()
+    fn process_send_take(args: account_parser::SendTakeArgs) -> DexResult {
+
+        let account_parser::SendTakeArgs {
+            instruction,
+            signer,
+            mut event_q,
+            mut req_q,
+            mut order_book_state,
+            coin_wallet,
+            pc_wallet,
+            coin_vault,
+            pc_vault,
+            spl_token_program,
+            fee_tier,
+            vault_signer
+        } = args;
+
+        // Question about architecture of max coin, max price, min, etc
+        let limit_price = instruction.limit_price;
+        let mut limit = instruction.limit;
+        let min_coin_qty = instruction.min_coin_qty;
+        let min_native_pc_qty = instruction.min_native_pc_qty;
+
+        let order_type = OrderType::ImmediateOrCancel;
+        let self_trade_behavior = SelfTradeBehavior::AbortTransaction;
+        let native_pc_qty_locked;
+
+        let order_id = req_q.gen_order_id(instruction.limit_price.get(), instruction.side); 
+        let mut proceeds = RequestProceeds::zero();
+        
+        match instruction.side {
```

Comment:
I'm personally a fan of doing:
```
let native_pc_qty_locked = match instruction.side {
    ...
};
```
Initialize and declare at the same time to make the code more legible